dashboard/app/views/levels/editors/_dsl.html.haml

```diff
@@ -6,3 +6,33 @@
   - cols = @dsl_file ? @dsl_file.lines.max_by(&:length).length : 20
   - rows = @dsl_file ? @dsl_file.lines.count + 1 : 10
   = text_area_tag('level[dsl_text]', @dsl_file || (@level.dsl_default), rows: rows, cols: cols, style: 'width:500px')
+
+%p Markdown Editor
+#epiceditor{style: 'height: 350px; border: 5px;'}
+= stylesheet_link_tag    'editor/markdown_editor', media: 'all'
+= javascript_include_tag 'editor/markdown_editor'
+
+:javascript
+  var dslElement = $('#level_dsl_text');
+  var dslText = dslElement.val();
+  // Match against markdown heredoc syntax and capture contents in [2].
+  var regex = /markdown <<(\w*)\n([\s\S]*)\n\1$/m;
```

Comment:
mostly equivalent with one difference: `.*` doesn't match on newlines in a multi-line match, so `[\s\S]*` is a suggested workaround I found.